src/main/java/dev/fastgql/sql/SQLQuery.java

```diff
@@ -80,7 +94,51 @@ public void reset() {
    */
   public String build() {
     return String.format(
-        "SELECT %s FROM %s %s %s %s",
-        String.join(", ", keys), table, alias, String.join(" ", joins), String.join(" ", suffixes));
+            "SELECT %s FROM %s %s %s %s",
+            String.join(", ", keys), table, alias, String.join(" ", joins), buildConstraints())
+        .replaceAll("\\s+", " ");
+  }
+
+  private String buildConstraints() {
+    return String.format(
+        "%s %s %s %s",
+        buildWhereQuery(), buildOrderByQuery(), buildLimitQuery(), buildOffsetQuery());
+  }
+
+  private String buildOrderByQuery() {
+    if (args.getOrderBy() == null || !args.getOrderBy().isJsonArray()) {
+      return "";
+    }
+    return String.format(
+        "ORDER BY %s", createOrderByQuery(args.getOrderBy().getAsJsonArray(), alias, fieldToAlias));
+  }
+
+  private String buildLimitQuery() {
+    if (args.getLimit() == null) {
+      return "";
+    }
+    return String.format("LIMIT %d", args.getLimit());
+  }
+
+  private String buildOffsetQuery() {
+    if (args.getOffset() == null) {
+      return "";
+    }
+    return String.format("OFFSET %d", args.getOffset());
+  }
+
+  private String buildWhereQuery() {
+    if (args.getWhere() != null && args.getWhere().isJsonObject()) {
+      String whereQuery = createBoolQuery(args.getWhere().getAsJsonObject(), alias, fieldToAlias);
+      addSuffix(String.format("(%s)", whereQuery));
```

Comment:
addSuffix is no longer good name for this function, should be something like addWhereConditional